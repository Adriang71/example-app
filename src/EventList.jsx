import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import ListItem from '@material-ui/core/ListItem';
import List from '@material-ui/core/List';
import Grid from '@material-ui/core/Grid';
import { useSelector, useDispatch } from 'react-redux'
import { loadEvent } from './Actions';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  pos: {
    marginBottom: 12,
  },
});

export default function EventList() {
  const isLoading = useSelector(state => state.isLoading)
  const events = useSelector(state => state.events)
  const dispatch = useDispatch()

  const classes = useStyles();
  const bull = <span className={classes.bullet}>â€¢</span>;

  React.useEffect(() => {
    dispatch(loadEvent())
  }, [dispatch]);

  if(isLoading) return <Grid item>Loading...</Grid>

  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
    >
    {isLoading && <div>Loading...</div>}
    {!isLoading && events.length > 0 &&
      events.reverse().map(event => (
        <List key={event.id}>
          <ListItem >
            <Card className={classes.root}>
              <CardContent>
                <Typography variant="h6" component="h3">
                  {event.name} {event.lastName}
                </Typography>
                <Typography className={classes.pos} color="textSecondary">

                  Event date {bull} {event.date}
                </Typography>
                <Typography variant="body2" component="p">
                  {event.email}
                </Typography>
              </CardContent>
              <CardActions>
                <Button size="small">Check Event</Button>
              </CardActions>
            </Card>
          </ListItem>
        </List>
      ))
    }
    </Grid>
  );
}