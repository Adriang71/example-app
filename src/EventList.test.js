import React from 'react'
import { render, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';
import fetchMock from 'fetch-mock'
import { Provider } from 'react-redux';
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import EventList from './EventList';

const TestProvider = ({store, children}) =>
  <MuiPickersUtilsProvider utils={MomentUtils}>
    <Provider store={store}>
      {children}
    </Provider>
  </MuiPickersUtilsProvider>

const mockStore = configureMockStore([thunk]);


describe("list should", () => {
  let emit;

  beforeAll(() => {
    ({ emit } = window._virtualConsole);
  });

  beforeEach(() => {
    fetchMock.post('*', {response: 200}, { overwriteRoutes: false })
    fetchMock.get('*', {response: 200}, { overwriteRoutes: false })
    window._virtualConsole.emit = jest.fn();
  });

  afterAll(() => {
    window._virtualConsole.emit = emit;
    fetchMock.restore();
  });

  it("display properly", () => {
    const store = mockStore({
        isLoading: false,
        events:[]
    });
    const { asFragment, container } = render(
      <TestProvider store={store}>
        <EventList />
      </TestProvider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(container).toMatchSnapshot();
  });

  it("display loading", async () => {
    const store = mockStore({
        isLoading: true,
        events:[]
    });
    const {  queryByText } = render(
      <TestProvider store={store}>
        <EventList />
      </TestProvider>
    );
    await wait(() => {
      expect(queryByText('Loading...')).not.toBeNull()
    });
  });

  it("not display loading", async () => {
    const store = mockStore({
        isLoading: false,
        events:[]
    });
    const {  queryByText } = render(
      <TestProvider store={store}>
        <EventList />
      </TestProvider>
    );
    await wait(() => {
      expect(queryByText('Loading...')).toBeNull()
    });
  });

  it("not display events", async () => {
    const store = mockStore({
        isLoading: false,
        events:[]
    });
    const {  queryByText } = render(
      <TestProvider store={store}>
        <EventList />
      </TestProvider>
    );
    await wait(() => {
      expect(queryByText('Check Event')).toBeNull()
    });
  });

  it("display events", async () => {
    const store = mockStore({
        isLoading: false,
        events:[
          {id:1, name: 'foo', lastName: 'Bar', email: 'foo@bar.pl', date: '22/22/22'},
          {id:2, name: 'foo', lastName: 'Bar', email: 'foo@bar.pl', date: '22/22/22'},
          {id:3, name: 'foo', lastName: 'Bar', email: 'foo@bar.pl', date: '22/22/22'}
        ]
    });
    const { queryAllByText } = render(
      <TestProvider store={store}>
        <EventList />
      </TestProvider>
    );
    await wait(() => {
      expect(queryAllByText('Check Event')).not.toBeNull()
    });
  });
});


