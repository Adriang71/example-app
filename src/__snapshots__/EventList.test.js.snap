// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should display properly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  />
</DocumentFragment>
`;

exports[`list should display properly 2`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  />
</div>
`;
